### DISCUSSION TITLE
How do i make the background transparent

### DISCUSSION THREAD

**Discussion by glepnir** [fltk-rs/fltk-rs#1266] [Q&A]
read the doc check the fltk-theme . I don't find the some informations about background transparent

**Comment by MoAlyousef** [comment]
You can try set_opacity():

https://docs.rs/fltk/latest/fltk/prelude/trait.WindowExt.html#tymethod.set_opacity

**Comment by glepnir** [comment]
@MoAlyousef  Thanks for help. it works but i want these area is transparent .can i do it?

![image](https://user-images.githubusercontent.com/41671631/182841172-a69f42b5-4f3b-49ab-8ff1-0a292250d6fa.png)

**Reply by MoAlyousef** [reply]
I don't think FLTK supports partial opacity.

**Reply by glepnir** [reply]
It will look weird if you use a circular image.

**Comment by glepnir** [comment]
another question how can i keep the cursor in terminal not into window. like  use  `path/to/binaryfile  some args`  this in terminal to start this gui .I want keep the cursor in termianl not into gui window.

**Reply by MoAlyousef** [reply]
You can lauch it detached:
```
path/to/binaryfile some args &
```

**Reply by glepnir** [reply]
oh get some error `cargo run /Users/joyce/Pictures/868059.png 400 400 &`
```
     Running `target/debug/imageview /Users/joyce/Pictures/868059.png 400 400`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("EOF while parsing a value", line: 1, column: 0)', src/main.rs:49:64
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

check the line I read some command line output and serialize to a struct in this line .

```rust
    let args: Vec<String> = env::args().collect();
    let filename = &args[1];
    let image_width_pixel: i32 = args[2].trim().parse().unwrap();
    let image_height_pixel: i32 = args[3].trim().parse().unwrap();

    // wezterm data
    let child = Command::new("wezterm")
        .args(&["cli", "list", "--format", "json"])
        .output()
        .expect("Failed to execute command");
    let output = String::from_utf8_lossy(&child.stdout);
    let wez_data: Vec<WezData> = serde_json::from_str(&output).unwrap();

```

**Reply by MoAlyousef** [reply]
To pass command line args to your binary when run thru cargo:
`cargo run — /Users/joyce/Pictures/868059.png 400 400`

**Reply by glepnir** [reply]
@MoAlyousef  Thanks for quick reply. you are not sleep ? well also have error through out

```
     Running `target/debug/imageview '—' /Users/joyce/Pictures/868059.png 400 400`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("EOF while parsing a value", line: 1, column: 0)', src/main.rs:49:64
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

**Reply by MoAlyousef** [reply]
It might be an error from serde_json. It would be better to replace unwraps with expect() and some text to see where the error is happening. In the sample code you’ve shared, you don’t appear to be using the parsed command-line args.

**Reply by glepnir** [reply]
yup I used it on function below. code in [there](https://github.com/glepnir/dashboard-imageview) . looks like if use `-` the serialize will faield.
